src/main/java/service/operation/ReturnOperation.java

```diff
@@ -0,0 +1,23 @@
+package service.operation;
+
+import data.db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class ReturnOperation implements OperationHandler {
+
+    @Override
+    public void handle(FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getQuantity() <= 0) {
+            return;
+        }
+        for (Map.Entry<String, Integer> fruit : Storage.getFruitsStorage().entrySet()) {
+            if (fruit.getKey().equals(fruitTransaction.getFruit())) {
+                Storage.setFruitsStorage(fruit.getKey(),
+                        fruit.getValue() + fruitTransaction.getQuantity());
+                return;
+            }
+        }
```

Comment:
you degraded HashMap's performance from constant time to linear - no need to iterate all the entries, that's not how we use hashmap, fix in every handler